.github/workflows/check-consistent-dependencies.yml

```diff
@@ -19,25 +19,27 @@ jobs:
     runs-on: ubuntu-24.04
 
     steps:
+      # Always checkout the code because we don't always have a PR url.
+      - uses: actions/checkout@v5
+
       # Only run remaining steps if there are changes to requirements/**
       - name: "Decide whether to short-circuit"
-        env:
-          GH_TOKEN: "${{ github.token }}"
-          PR_URL: "${{ github.event.pull_request.html_url }}"
         run: |
-          paths=$(gh pr diff "$PR_URL" --name-only)
-          echo $'Paths touched in PR:\n'"$paths"
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            BASE_SHA="${{ github.event.pull_request.base.sha }}"
+          else
+            BASE_SHA="${{ github.event.merge_group.base_sha }}"
+          fi
+
+          # Fetch the base sha so we can compare. It's not checked checkout by
+          # default.
+          git fetch origin "$BASE_SHA"
 
           # The ^"? is because git may quote weird file paths
-          matched="$(echo "$paths" | grep -P '^"?((requirements/)|(scripts/.*?/requirements/))' || true)"
-          echo $'Relevant paths:\n'"$matched"
-          if [[ -n "$matched" ]]; then
-              echo "RELEVANT=true" >> "$GITHUB_ENV"
+          if git diff --name-only "$BASE_SHA" | grep -P '^"?((requirements/)|(scripts/.*?/requirements/))'; then
+            echo "RELEVANT=true" >> "$GITHUB_ENV"
```

Comment:
We may want to include a comment about why this approach to short-circuiting is being used.